codeguru_profiler_agent/profiler_disabler.py

```diff
@@ -38,19 +42,39 @@ class CpuUsageCheck:
     def __init__(self, timer):
         self.timer = timer
 
-    def is_cpu_usage_limit_reached(self, profile=None):
+    # This function carries out an overall cpu limit check that covers the cpu overhead caused for the full
+    # sampling cycle: sample -> aggregate -> report -> refresh config. We expect this function to be called after
+    # configuration refresh and profile submission.
```

Comment:
to me the cycle is more "refresh config -> (sample -> aggregate) * n -> profile submission. See comment below but I think we should only check for overall overhead after profile submission.